domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Agency.java

```diff
@@ -0,0 +1,275 @@
+         */
+        public AgencyBuilder agencyUrl(@NotNull final String agencyUrl) {
+            this.agencyUrl = agencyUrl;
+            return this;
+        }
+
+        /**
+         * Sets field agencyTimezone value and returns this
+         *
+         * @param agencyTimezone timezone where the transit agency is located
+         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyTimezone(@NotNull final String agencyTimezone) {
+            this.agencyTimezone = agencyTimezone;
+            return this;
+        }
+
+        /**
+         * Sets field agencyId value and returns this
+         *
+         * @param agencyId identifies a transit brand which is often synonymous with a transit agency
+         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyId(@Nullable final String agencyId) {
+            this.agencyId = agencyId;
+            return this;
+        }
+
+        /**
+         * Sets field agencyLang value and returns this
+         *
+         * @param agencyLang primary language used by this transit agency
+         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyLang(@Nullable final String agencyLang) {
+            this.agencyLang = agencyLang;
+            return this;
+        }
+
+        /**
+         * Sets field agencyPhone value and returns this
+         *
+         * @param agencyPhone a voice telephone number for the specified agency
+         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyPhone(@Nullable final String agencyPhone) {
+            this.agencyPhone = agencyPhone;
+            return this;
+        }
+
+        /**
+         * Sets field agencyFareUrl value and returns this
+         *
+         * @param agencyFareUrl URL of a web page that allows a rider to purchase tickets or other fare instruments for
+         *                      that agency online         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyFareUrl(@Nullable final String agencyFareUrl) {
+            this.agencyFareUrl = agencyFareUrl;
+            return this;
+        }
+
+        /**
+         * Sets field agencyEmail value and returns this
+         *
+         * @param agencyEmail email address actively monitored by the agency’s customer service department
+         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyEmail(@Nullable final String agencyEmail) {
+            this.agencyEmail = agencyEmail;
+            return this;
+        }
+
+        /**
+         * Returns list of notice generated when building an {@code Agency} from this.
+         *
+         * @return the list of notice generated when building an {@link Agency} from this.
+         */
+        public List<Notice> getNoticeCollection() {
```

Comment:
From @fabrice-v : There should not be a getter here, instead using java error/success class could be a solution